eZ/Publish/Core/MVC/Symfony/Routing/UrlAliasRouter.php

```diff
@@ -64,13 +73,16 @@ public function __construct(
         URLAliasService $urlAliasService,
         ContentService $contentService,
         UrlAliasGenerator $generator,
+        Repository $repository,
         RequestContext $requestContext,
         LoggerInterface $logger = null
     ) {
         $this->locationService = $locationService;
         $this->urlAliasService = $urlAliasService;
         $this->contentService = $contentService;
         $this->generator = $generator;
+        $this->repository = $repository;
+        $this->permissionResolver = $this->repository->getPermissionResolver();
```

Comment:
Never do this. Pass `PermissionResolver` as another dependency and let container builder to inject it.

Reply:
+1, considering PermissionResolver is already decoupled and available as a service.